### PR TITLE
OPENEUROPA-2205: Create the event profile bundleable entity.

### PR DESCRIPTION
**Pull request by nagyad**

### CODE REVIEW

File: modules/oe_content_event/oe_content_event.permissions.yml [hunk 1]
```diff
@@ -1,3 +1,7 @@
+administer event profile entities:
+  title: 'Administer Event profiles'
```
Comment (sergepavle) [inline modules/oe_content_event/oe_content_event.permissions.yml]:
```suggestion
  title: 'Administer Event profiles entities'
```
?

File: modules/oe_content_event/src/Form/EventProfileTypeDeleteForm.php [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  public function buildForm(array $form, FormStateInterface $form_state) {
+    $num_lists = $this->queryFactory
+      ->get('event_profile')
+      ->condition('bundle', $this->entity
+        ->id())
+      ->count()
+      ->execute();
+    if ($num_lists) {
+      $caption = '<p>' . $this
+        ->formatPlural($num_lists, '%type is used by 1 event profile on your site. You can not remove this event profile type until you have removed the %type event profile.', '%type is used by @count event profile on your site. You may not remove %type until you have removed all of the %type event profiles.', [
```
Comment (sergepavle) [inline modules/oe_content_event/src/Form/EventProfileTypeDeleteForm.php]:
```suggestion
        ->formatPlural($num_lists,
        '%type is used by 1 event profile on your site. You can not remove this event profile type until you have removed the %type event profile.',
        '%type is used by @count event profile on your site. You may not remove %type until you have removed all of the %type event profiles.',
        [
```

File: modules/oe_content_event/src/Form/EventProfileTypeForm.php [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+    $entity_type = $this->entity;
+    $status = $entity_type->save();
+    $message_params = [
+      '%label' => $entity_type->label(),
+      '%content_entity_id' => $entity_type->getEntityType()->getBundleOf(),
+    ];
+
+    switch ($status) {
+      case SAVED_NEW:
+        drupal_set_message($this->t('Created the %label %content_entity_id entity type.', $message_params));
```
Comment (sergepavle) [inline modules/oe_content_event/src/Form/EventProfileTypeForm.php]: $this->messenger()->

File: modules/oe_content_event/src/Form/EventProfileTypeForm.php [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+        break;
+
+      default:
+        drupal_set_message($this->t('Saved the %label %content_entity_id entity type.', $message_params));
```
Comment (sergepavle) [inline modules/oe_content_event/src/Form/EventProfileTypeForm.php]: $this->messenger()->

File: modules/oe_content_event/src/Form/EventProfileTypeDeleteForm.php [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static($container->get('entity.query'));
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
```
Comment (upchuk) [inline modules/oe_content_event/src/Form/EventProfileTypeDeleteForm.php]: This method has been formatted way too much :) Let's normalize a bit and not have every word on one line.

File: modules/oe_content_event/src/Form/EventProfileTypeForm.php [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+    return $this->protectBundleIdElement($form);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function actions(array $form, FormStateInterface $form_state) {
+    $actions = parent::actions($form, $form_state);
+
+    if (\Drupal::moduleHandler()->moduleExists('field_ui') && $this->getEntity()->isNew()) {
```
Comment (upchuk) [inline modules/oe_content_event/src/Form/EventProfileTypeForm.php]: Let's inject this.

File: modules/oe_content_event/src/Form/EventProfileTypeForm.php [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+
+  /**
+   * Form submission handler to redirect to Manage fields page of Field UI.
+   *
+   * @param array $form
+   *   The Event profile type form.
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   The form state.
+   */
+  public function redirectToFieldUi(array $form, FormStateInterface $form_state) {
```
Comment (upchuk) [inline modules/oe_content_event/src/Form/EventProfileTypeForm.php]: return type

**Pull request merged**